### ISSUE TITLE
tickTextOffset with negative value no longer displays tick labels

### ISSUE THREAD

**Issue by mayofaulkner** [pyqtgraph/pyqtgraph#1744]
### Short description

I have been using pyqtgraph to display different regions and was using the tickTextOffset with negative values in order to overlay my tick labels on the regions I was displaying.  After upgrading to version 0.11.1, tickTextOffset with negative values no longer works . I have looked into the code and found that this [line](https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/AxisItem.py#L1169) was introduced which clips the axis displayed and so changes the behaviour between version 0.11.0 and 0.11.1. Do you have any suggestions for how I can get around this? Many thanks for your help!

### Code to reproduce

```python
import pyqtgraph as pg
from pyqtgraph import QtGui
import numpy as np
pg.setConfigOption('background', 'w')
pg.setConfigOption('foreground', 'k')


plotWidget = pg.plot()
values = [[0, 100], [100, 200], [300, 500]]
cols = [[0, 100, 0, 255], [0, 150, 0, 255], [0, 200, 0, 255]]
labels = [(50, 'First Region'), (150, 'Second Region'), (400, 'Third Region')]

axis = plotWidget.plotItem.getAxis('left')
axis.setTicks([labels])
axis.setZValue(10)
axis.setStyle(tickTextOffset=-100) 

# Plot each region
for reg, col in zip(values, cols):
    colour = QtGui.QColor(*col)
    region = pg.LinearRegionItem(values=(reg[0], reg[1]),
                                 orientation=pg.LinearRegionItem.Horizontal,
                                 brush=colour, movable=False)
    plotWidget.plotItem.addItem(region)


pg.mkQApp().exec_()


```

### Previous behavior
Using pyqtgraph==0.11.0 I get the following image from the code above
![example_with_pyqtgraph0110](https://user-images.githubusercontent.com/11440465/116276226-3c09f180-a77c-11eb-981f-12a9a2216f68.png)

### Current behavior
With pyqtgraph >=0.11.1, the labels are not displayed
![example_with_pyqtgraph0111](https://user-images.githubusercontent.com/11440465/116276570-8d19e580-a77c-11eb-8544-b1a21f838cec.png)

### Tested environment(s)

 * PyQtGraph version: 0.11.0 and 0.11.1
 * Qt Python binding: PyQt5 5.15.4  PyQt5-Qt5 5.15.2
 * Python version: 3.7.0
 * NumPy version: 1.19.2
 * Operating system: Windows 10
 * Installation method: pip

**Comment by NilsNemitz** [comment]
Dear @mayofaulkner , sorry for the slow reply.

I have to admit that this is probably not an issue we will work on fixing with high priority. It should be possible to achieve the same effect with text labels.

However, there is this *one weird trick* that you might not know about:
If you add
``` python
plotWidget.plotItem.showGrid(False,True,0)
```
to your code, your labels should be displayed as before.

The reason is that in PyQtGraph, it is the AxisItems that draw the grid lines. So if you enable the grid for the y-axis, the clipping area for everything the vertical axis wants to draw is expanded to include the plotarea. Even if you set the grid opacity to zero so that nothing is actually visible.

I hope that might help you out until we get around to looking at this again. The idea of moving some axis info into the plot area might be quite useful for applications that value information density over everything else.

**Comment by NilsNemitz** [comment]
If that doesn't work because we haven't fixed issue #1484 yet, then try
```python
plotWidget.plotItem.getAxis('left').setGrid(0)
```

**Comment by mayofaulkner** [comment]
@NilsNemitz Thank you so much for your reply.  That works nicely. Many thanks!
